recipes/gdcm/all/conanfile.py

```diff
@@ -0,0 +1,228 @@
+            set(GDCM_BUILD_SHARED_LIBS "{build_shared_libs}")
+            
+            set(GDCM_USE_VTK "OFF")
+
+            # The "use" file.
+            set(GDCM_USE_FILE ${{SELF_DIR}}/UseGDCM.cmake)
+            
+            # The VTK options.
+            set(GDCM_VTK_DIR "")
+            
+            get_filename_component(GDCM_INCLUDE_ROOT "${{SELF_DIR}}/../../include/{self._gdcm_subdir}" ABSOLUTE)
+            set(GDCM_INCLUDE_DIRS ${{GDCM_INCLUDE_ROOT}})
+            get_filename_component(GDCM_LIB_ROOT "${{SELF_DIR}}/../../lib" ABSOLUTE)
+            set(GDCM_LIBRARY_DIRS ${{GDCM_LIB_ROOT}})
+        """)
+        tools.save(variables_file, content)
+
+    @staticmethod
+    def _create_cmake_module_alias_targets(targets_file, libraries):
+        content = "\n".join([textwrap.dedent(f"""\
+            if(TARGET GDCM::{l} AND NOT TARGET {l})
+                add_library({l} INTERFACE IMPORTED)
+                set_property(TARGET {l} PROPERTY INTERFACE_LINK_LIBRARIES GDCM::{l})
+            endif()
+            """) for l in libraries])
+        tools.save(targets_file, content)
+
+    @property 
+    def _gdcm_subdir(self):
+        v = tools.Version(self.version)
+        return f"gdcm-{v.major}.{v.minor}"
+
+    @property
+    def _gdcm_builddir(self):
+        return os.path.join("lib", self._gdcm_subdir)
+
+    @property
+    def _gdcm_cmake_module_aliases_path(self):
+        return os.path.join("lib", self._gdcm_subdir, f"conan-official-{self.name}-targets.cmake")
+
+    @property
+    def _gdcm_cmake_variables_path(self):
+        return os.path.join("lib", self._gdcm_subdir, f"conan-official-{self.name}-variables.cmake")
+
+    @property
+    def _gdcm_build_modules(self):
+        return [self._gdcm_cmake_module_aliases_path, self._gdcm_cmake_variables_path]
+
+    @property
+    def _gdcm_libraries(self):
+        gdcm_libs = ["gdcmcharls",
+                     "gdcmCommon",
+                     "gdcmDICT",
+                     "gdcmDSED",
+                     "gdcmexpat",
+                     "gdcmIOD",
+                     "gdcmjpeg12",
+                     "gdcmjpeg16",
+                     "gdcmjpeg8",
+                     "gdcmMEXD",
+                     "gdcmMSFF",
+                     "gdcmopenjp2",
+                     "socketxx"]
+        if self.settings.os == "Windows":
+            gdcm_libs.append("gdcmgetopt")
+        else:
+            gdcm_libs.append("gdcmuuid")
+        return gdcm_libs
+
+    def package_info(self):
+        self.cpp_info.names["cmake_find_package"] = "GDCM"
+        self.cpp_info.names["cmake_find_multi_package"] = "GDCM"
+        gdcm_libs = self._gdcm_libraries
+        self.cpp_info.components["headers"].includedirs = [os.path.join("include", self._gdcm_subdir)]
+        self.cpp_info.components["headers"].builddirs = [self._gdcm_builddir] 
+        self.cpp_info.components["headers"].build_modules["cmake"] = self._gdcm_build_modules
```

Comment:
Not use to seeing this 🤔

Reply:
I've removed headers component as it's not official part. Does conan have some way to depend on includes/defines without linking to subcomponents?